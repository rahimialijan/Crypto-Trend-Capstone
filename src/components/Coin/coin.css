.coin-container {
  display: flex;
  flex-direction: column;
}

.coin-container > span {
  display: flex;
  flex-direction: column;
}

@media screen and (max-width: 600px) {
  .coin-info {
    display: flex;
    gap: 2rem;
    padding: 1% 2%;
  }

  .coin-market-cap {
    display: none;
  }
}

 .coin-name-value, .coin-price-value{
  margin-left: 100px;
}
.coin-market-cap-value{
  margin-left: 52px;
}