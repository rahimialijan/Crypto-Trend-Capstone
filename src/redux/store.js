import { configureStore } from "@reduxjs/toolkit";
import cryptoReducer from "./crypto/cryptoSlice";
const REDUX_DEVTOOLS = true;

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
  devTools: REDUX_DEVTOOLS || false,
});

export default store;
