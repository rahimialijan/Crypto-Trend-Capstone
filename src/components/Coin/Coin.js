import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './coin.css';

const Coin = ({ coin }) => {
  const style = {
    backgroundImage: `url(${coin.iconUrl})`,
    backgroundRepeat: 'no-repeat',
    backgroundPositionX: 'right',
    backgroundSize: '50px',
  };
  return (
    <li key={coin.uuid}>
      <Link to={`details?coin=${coin.uuid}`} state={coin.uuid}>
        <div className="coin-container">
          <span style={style}>
            <span className="coin-info coin-name">
              <span className="coin-name-label">Name: </span>
              <span className="coin-name-value">{`${coin.name}(${coin.symbol})`}</span>
            </span>
            <span className="coin-info coin-market-cap">
              <span className="coin-market-cap-label">Market Cap: </span>
              <span className="coin-market-cap-value">
                {Number(coin.marketCap).toLocaleString()}
              </span>
            </span>
            <span className="coin-info coin-price">
              <span className="coin-price-label">Price: </span>
              <span className="coin-price-value">
                {Number(coin.price).toLocaleString()}
              </span>
            </span>
          </span>
        </div>
      </Link>
    </li>
  );
};

Coin.propTypes = {
  coin: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
};

export default Coin;
