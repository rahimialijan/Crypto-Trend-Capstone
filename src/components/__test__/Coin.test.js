import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Coin from '../Coin/Coin';
import store from '../../redux/store';

describe('', () => {
  test('Should Render Coin Component', () => {
    const mockCoin = {
      uuid: 111, symbol: '', name: 'BTC', iconUrl: '', marketCap: 100, price: 1000,
    };
    const container = render(
      <BrowserRouter>
        <Provider store={store}>
          <Coin coin={mockCoin} />
        </Provider>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
