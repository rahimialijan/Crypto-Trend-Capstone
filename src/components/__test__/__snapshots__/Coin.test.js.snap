// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Should Render Coin Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <a
          href="/details?coin=111"
        >
          <div
            class="coin-container"
          >
            <span
              style="background-image: url(); background-repeat: no-repeat; background-position-x: right; background-size: 50px;"
            >
              <span
                class="coin-info coin-name"
              >
                <span
                  class="coin-name-label"
                >
                  Name: 
                </span>
                <span
                  class="coin-name-value"
                >
                  BTC()
                </span>
              </span>
              <span
                class="coin-info coin-market-cap"
              >
                <span
                  class="coin-market-cap-label"
                >
                  Market Cap: 
                </span>
                <span
                  class="coin-market-cap-value"
                >
                  100
                </span>
              </span>
              <span
                class="coin-info coin-price"
              >
                <span
                  class="coin-price-label"
                >
                  Price: 
                </span>
                <span
                  class="coin-price-value"
                >
                  1,000
                </span>
              </span>
            </span>
          </div>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <a
        href="/details?coin=111"
      >
        <div
          class="coin-container"
        >
          <span
            style="background-image: url(); background-repeat: no-repeat; background-position-x: right; background-size: 50px;"
          >
            <span
              class="coin-info coin-name"
            >
              <span
                class="coin-name-label"
              >
                Name: 
              </span>
              <span
                class="coin-name-value"
              >
                BTC()
              </span>
            </span>
            <span
              class="coin-info coin-market-cap"
            >
              <span
                class="coin-market-cap-label"
              >
                Market Cap: 
              </span>
              <span
                class="coin-market-cap-value"
              >
                100
              </span>
            </span>
            <span
              class="coin-info coin-price"
            >
              <span
                class="coin-price-label"
              >
                Price: 
              </span>
              <span
                class="coin-price-value"
              >
                1,000
              </span>
            </span>
          </span>
        </div>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
